Update implementation

electerious opened the pull request:
This is just a personal preference, but I won't combine everything into AckeeTraker. It basically does two things when looking at the `if`s. They only share `trackerInstance.record()`.

Attention: It's not tested, but it's basically just a restructure.

electerious committed:
Update implementation

This is just a personal preference, but I won't combine everything into AckeeTraker. It basically does two things when looking at the `if`s. They only share `trackerInstance.record()`.

Attention: It's not tested, but it's basically just a restructure.

Burnsy commented:
I just ran this code in one of my gatsby websites and it breaks 
![Screenshot 2019-11-04 at 11 50 17](https://user-images.githubusercontent.com/15834048/68118958-49cd1680-fef9-11e9-99d6-dbb8b578110c.png)

As the tracker has not been created yet would be my guess. So it has to be wrapped in an if statement  to check if the tracker has been created yet. .

electerious committed:
Create instance first

`onRouteUpdate` seems to run before `onInitialClientRender` which means we can combine them and don't need `onInitialClientRender` at all.

electerious commented:
`onRouteUpdate` seems to run before `onInitialClientRender` which means we can combine them and don't need `onInitialClientRender` at all. This should hopefully do the job :)

Burnsy commented:
Sorry for the massive delay! Will check this out tonight and merge with the latest version aswell.

Burnsy commented:
Adding the package using [Pika](https://github.com/Burnsy/gatsby-plugin-ackee-tracker/pull/5/checks?check_run_id=295623458). it seems still does not work. it says that trackerInstance.record is not a function. Looking at the code it seems to be running trackerInstance before it is initialised (still false).

electerious commented:
That shouldn't be possible in the current implementation: https://github.com/Burnsy/gatsby-plugin-ackee-tracker/blob/9340c454d0da1eb4b1e6ac64d5c3e07b88bf572c/src/gatsby-browser.js

It will always create the instance first. I'm not sure where this error comes from.
